pkg/remoteexec/kube/target.go

```diff
@@ -0,0 +1,201 @@
+//
+// Copyright (c) 2018 The heketi Authors
+//
+// This file is licensed to you under your choice of the GNU Lesser
+// General Public License, version 3 or any later version (LGPLv3 or
+// later), or the GNU General Public License, version 2 (GPLv2), in all
+// cases as published by the Free Software Foundation.
+//
+
+package kube
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/heketi/heketi/pkg/kubernetes"
+)
+
+type describable interface {
+	String() string
+}
+
+// target contains attributes common to all target types.
+type target struct {
+	Namespace string
+}
+
+// TargetLabel represents resources in the cluster where the
+// specified key and value parameters are equal. These can
+// be used to determine a pod with a given label.
+type TargetLabel struct {
+	target
+	Key   string
+	Value string
+}
+
+// String describes the target label.
+func (t TargetLabel) String() string {
+	return fmt.Sprintf("label:%v=%v", t.Key, t.Value)
+}
+
+// GetTargetPod uses the target label's Key and Value parameters
+// to determine a unique matching pod. If no pod with the
+// matching label is found an error is returned. If more than one
+// pod with the matching label is found an error is returned.
+func (t TargetLabel) GetTargetPod(k *KubeConn) (TargetPod, error) {
+	var (
+		tp  TargetPod
+		err error
+		ns  = t.Namespace
+	)
+	if ns == "" {
+		ns, err = kubernetes.GetNamespace()
+		if err != nil {
+			return tp, fmt.Errorf("Namespace must be provided in configuration: %v", err)
+		}
+	}
+	tp.Namespace = ns
+	tp.origin = t
+	// Get a list of pods
+	pods, err := k.kube.Core().Pods(ns).List(v1.ListOptions{
+		LabelSelector: t.Key + "==" + t.Value,
+	})
+	if err != nil {
+		k.logger.Err(err)
+		return tp, fmt.Errorf("Failed to get list of pods")
+	}
+
+	numPods := len(pods.Items)
+	if numPods == 0 {
+		// No pods found with that label
+		err := fmt.Errorf("No pods with the label '%v=%v' were found",
+			t.Key, t.Value)
+		k.logger.Critical(err.Error())
+		return tp, err
+
+	} else if numPods > 1 {
+		// There are more than one pod with the same label
+		err := fmt.Errorf("Found %v pods with the sharing the same label '%v=%v'",
```

Comment:
if we log the pod names will it be helpful?

Reply:
done in followup commit